src/libs/metadata.c

```diff
@@ -324,100 +459,243 @@ void connect_key_accels(dt_lib_module_t *self)
+    gtk_text_buffer_create_tag (buffer, "italic", "style", PANGO_STYLE_ITALIC, NULL);
+    GtkWidget *textview = gtk_text_view_new_with_buffer(buffer);
+    gtk_container_add(GTK_CONTAINER(swindow), textview);
+
+    gtk_text_view_set_wrap_mode(GTK_TEXT_VIEW(textview), GTK_WRAP_WORD);
+    gtk_text_view_set_accepts_tab(GTK_TEXT_VIEW(textview), FALSE);
+    dt_gui_key_accel_block_on_focus_connect(textview);
+    g_signal_connect(textview, "key-press-event", G_CALLBACK(key_pressed), self);
+    g_signal_connect(G_OBJECT(textview), "button-press-event", G_CALLBACK(click_on_textview), self);
+    g_signal_connect(textview, "grab-focus", G_CALLBACK(got_focus), self);
+    g_signal_connect(G_OBJECT(swindow), "scroll-event", G_CALLBACK(mouse_scroll), self);
+    d->textview[i] = GTK_TEXT_VIEW(textview);
+    gtk_widget_set_hexpand(textview, TRUE);
+    gtk_widget_set_vexpand(textview, TRUE);
+
+    // doesn't work. Workaround => gui_post_expose
+    // gtk_widget_set_no_show_all(GTK_WIDGET(label), TRUE);
+    // gtk_widget_set_no_show_all(GTK_WIDGET(textview), TRUE);
+  }
 
-    g_signal_connect(entry, "key-press-event", G_CALLBACK(key_pressed), self);
+  PangoLayout *cell = gtk_widget_create_pango_layout(GTK_WIDGET(d->textview[0]), "X");
 
-    gtk_entry_set_width_chars(GTK_ENTRY(entry), 0);
+  pango_layout_get_size(cell, NULL, &d->line_height);
+  g_object_unref(cell);
+  d->line_height /= PANGO_SCALE;
 
-    gtk_grid_attach(GTK_GRID(self->widget), label, 0, line, 1, 1);
-    gtk_grid_attach_next_to(GTK_GRID(self->widget), combobox, label, GTK_POS_RIGHT, 1, 1);
-  }
+  d->init_layout = FALSE;
 
   // clear/apply buttons
 
-  grid = GTK_GRID(gtk_grid_new());
+  grid = (GtkGrid *)gtk_grid_new();
   gtk_grid_set_column_homogeneous(grid, TRUE);
 
-  button = gtk_button_new_with_label(_("clear"));
+  GtkWidget *button = gtk_button_new_with_label(_("clear"));
   d->clear_button = button;
   gtk_widget_set_tooltip_text(button, _("remove metadata from selected images"));
-  gtk_grid_attach(grid, button, 0, 0, 1, 1);
-  g_signal_connect(G_OBJECT(button), "clicked", G_CALLBACK(clear_button_clicked), (gpointer)self);
+  gtk_grid_attach(grid, button, 0, 0, 5, 1);
+  g_signal_connect(G_OBJECT(button), "clicked", G_CALLBACK(clear_button_clicked), self);
 
   button = gtk_button_new_with_label(_("apply"));
   d->apply_button = button;
   gtk_widget_set_tooltip_text(button, _("write metadata for selected images"));
-  gtk_grid_attach(grid, button, 1, 0, 1, 1);
-  g_signal_connect(G_OBJECT(button), "clicked", G_CALLBACK(apply_button_clicked), (gpointer)self);
+  gtk_grid_attach(grid, button, 5, 0, 5, 1);
+  g_signal_connect(G_OBJECT(button), "clicked", G_CALLBACK(apply_button_clicked), self);
+
+  button = dtgtk_button_new(dtgtk_cairo_paint_preferences,
+      CPF_DO_NOT_USE_BORDER | CPF_STYLE_BOX | CPF_BG_TRANSPARENT, NULL);
+  d->config_button = button;
+  gtk_widget_set_tooltip_text(button, _("configure metadata"));
+  gtk_grid_attach(grid, button, 10, 0, 1, 1);
```

Comment:
10 replaced by 6.

And the we can have a small panel.

Reply:
I did this to reduce the space used by the configuration button. How can keep the configuration space as reduced as possible ?

Reply:
Maybe by using the same kind of code we have in the export module ?

Reply:
in export module boxes are used instead of grid.
With you data I get this:
![image](https://user-images.githubusercontent.com/23012047/74285488-b92a8180-4d04-11ea-9476-b8c03dd770c4.png)

Reply:
got it:
![image](https://user-images.githubusercontent.com/23012047/74286311-43bfb080-4d06-11ea-8d9f-8589e2a95396.png)

Reply:
and no need 3 & 3 neither. 1 & 1 works.

Reply:
Thanks for challenging ! :)

Reply:
![image](https://user-images.githubusercontent.com/23012047/74286611-e4ae6b80-4d06-11ea-9566-5e56fd6b0683.png)

Reply:
![image](https://user-images.githubusercontent.com/23012047/74286632-f42db480-4d06-11ea-9964-505e3611465c.png)